force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -0,0 +1,58 @@
+public class TaskTriggerHandler {
+
+    public static void updateRelatedAccounts(List<Task> tasks){
+        
+      Set<Id> accIds = new Set<Id>();
+  	  for (Task task : tasks){
+          if (task.whatId !=null && String.valueOf(task.whatId).startsWith('001')) {
+              accIds.add(task.whatId);
```

Comment:
Can I ever have a Task without WhatId ? 

What would happen if we did not check for 001 ?

Reply:
Yes, I can have a task without WhatId because this field is not required. 
The task can be related not only to Accounts (whose ids start with '001'), but also to another objects like Opportunities (whose ids start with '006'), Cases etc. In my case I only need the tasks related to Accounts.

Also there are task related to a contact id's or a lead's id, than they will have a WhoId.